build_pkg.sh

```diff
@@ -0,0 +1,43 @@
+#! /bin/bash
+
+set -e
+
+if [ $# -lt 2 ]; then
+    echo "Usage: $0 <package_dir> <version> [.deb files]"
+    echo ""
+    echo "Example: $0 px4-ros2-interface-lib/px4_ros2_cpp 0.0.100 path/to/ros-humble-px4-msgs_0.0.100-0jammy_arm64.deb"
+    exit 1
+fi
+
+WORK_DIR=$(realpath "$1")
+shift
+VERSION="$1"
+shift
+
+# Replace version in package.xml
+PACKAGE_XML="$WORK_DIR"/package.xml
+[ ! -f "$PACKAGE_XML" ] && echo "File $PACKAGE_XML is missing" && exit 1
+sed -i "s|<version>.*</version>|<version>$VERSION</version>|" "$PACKAGE_XML"
+
+# Copy deb files to a temporary directory & update the package dependencies
+DEB_TMP_DIR="$(mktemp -d)"
+trap 'rm -rf "$DEB_TMP_DIR"' EXIT
+for deb in "$@"; do
+    cp "$deb" "$DEB_TMP_DIR"
+
+    # Extract '<pkg-name>' from 'ros-<ros-version>-<pkg-name>_0.0.1-0jammy_arm64.deb'
+    ros_pkg_version="$(basename "$deb" | sed -n 's/ros-\([^-]*\)-\(.*\)_.*.deb/\2/p')"
+    ros_pkg=$(echo "$ros_pkg_version" | sed -n 's/\(.*\)_\([^_]*\)/\1/p')
+    [ -z "$ros_pkg" ] && echo "Failed to extract ros package name from $deb" && exit 1
+
+    # Replace <pkg-name>=<version> with just <pkg-name> in all rosdep files, since we inject this package.
+    # The drawback is that the final .deb will not depend on the specific version of <pkg-name>.
+    for rosdep in "$WORK_DIR"/rosdep-*.yaml; do
+      sed -i "s|$ros_pkg=.*|$ros_pkg|" "$rosdep"
```

Comment:
@bkueng  Is there any reason to avoid making the substitution in the rosdep.yaml include the version of the debian we're injecting?
I experimented with the following. I'm happy to push that, unless you see a reason against it
```suggestion
    read ros_pkg ros_version <<< "$(echo "$ros_pkg_version" | sed -n 's/\(.*\)_\([^_]*\)/\1 \2/p')"
    [ -z "$ros_pkg" ] && echo "Failed to extract ros package name from $deb" && exit 1
    [ -z "$ros_version" ] && echo "Failed to extract ros package version from $deb" && exit 1

    # Replace <pkg-name>[=<any-version>] with <pkg-name>=<ros-version> in all rosdep files, since we inject this package.
    for rosdep in "$WORK_DIR"/rosdep-*.yaml; do
      sed -i "s|$ros_pkg\(=.*\)\?|$ros_pkg=$ros_version|" "$rosdep"
```

Reply:
Yes because I thought `install_rosdeps.py` might again replace it from a cloudsmith version if that also exists.
I did not verify that, so if we know that is not the case, we can certainly set the version.

Reply:
I did some testing and it turns out apt will just check the debian version, not the contents nor the hash of the package. So it will not pull from any apt repo if a pinned package is already installed with the same version, despite different contents (unless the `--reinstall` flag is specified).

Within a build env container to build auterion-sdk:
1. Install local px4_msgs deb
```bash
root@7e1fa23ebff3:/work# dpkg -i /deb/ros-humble-px4-msgs_0.0.1-0jammy_arm64.deb
```
2.  Run the rosdep_install.py script
```bash
root@7e1fa23ebff3:/work# /scripts/install_rosdeps.py /work/rosdep-humble.yaml 
...
Manually installing the following packages via APT, before rosdep scan:
  - ros-humble-px4-msgs=0.0.1-0jammy
  - ros-humble-px4-ros2-cpp=0.0.1-0jammy
  - ros-humble-auterion-core-msgs=0.0.5-0jammy
...
ros-humble-px4-msgs is already the newest version (0.0.1-0jammy).
The following NEW packages will be installed:
  ros-humble-auterion-core-msgs ros-humble-px4-ros2-cpp
```

px4_msgs does not get reinstalled, other packages get pulled from cloudsmith.

I'll go ahead and push this then.

Reply:
Hmm nevermind, I was wrong. For some reason px4-ros2-cpp doesn't show the same behavior:

```
root@adf8551e5e7e:/work# dpkg -i /deb/ros-humble-px4-msgs_0.0.1-0jammy_arm64.deb /deb2/ros-humble-px4-ros2-cpp_0.0.1-0jammy_arm64.deb 
...

root@adf8551e5e7e:/work# /scripts/install_rosdeps.py /work/rosdep-humble.yaml 
...
Manually installing the following packages via APT, before rosdep scan:
  - ros-humble-px4-ros2-cpp=0.0.1-0jammy
  - ros-humble-px4-msgs=0.0.1-0jammy
  - ros-humble-auterion-core-msgs=0.0.5-0jammy
...
ros-humble-px4-msgs is already the newest version (0.0.1-0jammy).
The following NEW packages will be installed:
  ros-humble-auterion-core-msgs
The following packages will be upgraded:
  ros-humble-px4-ros2-cpp
...
Unpacking ros-humble-px4-ros2-cpp (0.0.1-0jammy) over (0.0.1-0jammy) ...
```
px4_msgs doesn't get overwritten despite a remote 0.0.1 being available, but px4-ros2-cpp does 🤷🏻‍♂️ 

I'll revert my changes in the meantime

Reply:
Thanks for testing. Should we merge this then as is?

Reply:
Sounds good to me 👍🏻